specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,592 @@
+      [out] EventRegistrationToken* token);
+  /// Remove listener for start stream request.
+  HRESULT remove_StartRequested(
+      [in] EventRegistrationToken token);
+  /// Listen to stop stream request once the stream started.
+  /// It is called when user stop all streaming requests from
+  /// the renderers (Javascript) or the host calls the Stop API. The renderer
+  /// can stream again by calling the streaming request API.
+  /// The renderer cleared all registered buffers before sending
+  /// the stop request event so that the callback of the next start request
+  /// should register the textures again.
+  /// The event is triggered when all requests for given stream id closed
+  /// by the Javascript, or the host's Stop API call.
+  HRESULT add_StopRequested(
+      [in] ICoreWebView2StagingTextureStreamStopRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for stop stream request.
+  HRESULT remove_StopRequested(
+      [in] EventRegistrationToken token);
+  /// Creates shared buffer that will be referenced by the host and the browser.
+  /// By using the shared buffer mechanism, the host does not have to
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal buffer id, which is the identity of the shared buffer.
+  /// The shared buffer is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableBuffer,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableBuffer returns an error, then the host calls the
+  /// CreateBuffer to allocate new shared buffer.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+  HRESULT CreateBuffer(
+    [in] UINT32 width,
+    [in] UINT32 height,
+    [out, retval] ICoreWebView2StagingTexture** buffer);
+  /// GetAvailableBuffer can be called on any thread like SetBuffer.
+  HRESULT GetAvailableBuffer([out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Remove texture buffer when the host removes the backed 2D texture.
+  /// The host can save the existing resources by deleting 2D textures
+  /// when it changes the frame sizes.
+  HRESULT RemoveBuffer([in] ICoreWebView2StagingTexture* buffer);
+  /// Indicates that the buffer is ready to present.
+  /// The buffer must be retrieved from the GetAvailableBuffer.
+  /// The host writes new texture to the local shared 2D texture of
+  /// the buffer id, which is created via CreateBuffer.
+  /// SetBuffer API can be called in any thread.
+  HRESULT SetBuffer([in] ICoreWebView2StagingTexture* buffer,
+    [in] ULONGLONG timestamp);
+  /// Render texture that is current set ICoreWebView2StagingTexture.
+  HRESULT Present();

```

Comment:
Is there a reason why SetBuffer and Present are separate? Present renders the texture that was last set via SetBuffer (right?). Should they be combined and Present should take the SetBuffer parameters? Or is there a reason to call SetBuffer and then wait to call Present?

Reply:
It is feedback from the Edge graphics side code reviewer; Present is an API that DirectX to use for commit rendering, and SetBuffer is kind of Swap to front buffer. It is said that it is familiar pattern with the graphics developer.

Reply:
> I've addressed comments, please reactivate anything it isn't clear to you. Thanks a lot.

Some of my comments are still unresolved. For example:
 * https://github.com/MicrosoftEdge/WebView2Feedback/pull/2743/files/bc5d2674edecfdf303bdf8d28308a44a61eee0be#r1020474503
 * https://github.com/MicrosoftEdge/WebView2Feedback/pull/2743/files/bc5d2674edecfdf303bdf8d28308a44a61eee0be#r1020472689
 * https://github.com/MicrosoftEdge/WebView2Feedback/pull/2743/files/bc5d2674edecfdf303bdf8d28308a44a61eee0be#r1020472927

Is that intentional?

Reply:
That's great. We'll just need to ensure that the documentation is clear.